doc/coq.rst

```diff
@@ -0,0 +1,494 @@
+  install under the ``user-contrib`` prefix the ``.cmxs`` files that appear in
+  ``<ocaml_libraries>``,
+
+- ``<coq_flags>`` will be passed to ``coqc`` as command-line options.
+  ``:standard`` is taken from the value set in the ``(coq (flags <flags>))``
+  field in ``env`` profile. See :ref:`dune-env` for more information.
+
+- The path to the ``<ocaml_libraries>`` installed locations will be passed to
+  ``coqdep`` and ``coqc`` using Coq's ``-I`` flag; this allows for a Coq theory
+  to depend on a ML plugin,
+
+- Your Coq theory can depend on other theories by specifying them in the
+  ``<coq_theories>`` field. Dune will then pass to Coq the corresponding flags
+  for everything to compile correctly [ ``-Q`` ]. As of today, we only support
+  composition with libraries defined in the same scope (that is to say, under
+  the same ``dune-project`` domain). We will lift this restriction in the
+  future. Note that composition with the Coq's standard library is supported,
+  but in this case the ``Coq`` prefix will be made available in a qualified way,
+  since Coq lang 0.2.
+
+- You can enable the production of Coq's native compiler object files by setting
+  ``<coq_native_mode>`` to ``native``. This will pass ``-native-compiler on`` to
+  Coq and install the corresponding object files under ``.coq-native``, when in
+  ``release`` profile. The regular ``dev`` profile will skip native compilation
+  to make the build faster, available since Coq v0.3.
+
+  Please note: support for native compute is **experimental**, and requires Coq
+  >= 8.12.1; moreover, depending libraries *must* be built with ``(mode
+  native)`` for this to work. Also, Coq must be configured to support native
+  compilation. Note that Dune will explicitly disable output of native
+  compilation objects when ``(mode vo)``, even if the default Coq's configure
+  flag enabled it. This will be improved in the future.
+
+Coq Documenation
+~~~~~~~~~~~~~~~~
+
+Given a :ref:`coq-theory` stanza with ``name A``, dune will produce two
+*directory targets*, ``A.html/`` and ``A.tex/``. HTML or LaTeX documenation for
+a Coq theory may then be built by running ``dune build A.html`` or ``dune build
+A.tex`` respectively (if the dune file for the theory is the current directory).
+
+There are also two aliases that can be called, which will build the HTML or
+LaTeX documentation and these are ``@doc`` and ``@doc-latex`` respectively.
+
+.. _include-subdirs-coq:
+
+Recursive Qualification of Modules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you add:
+
+.. code:: scheme
+
+    (include_subdirs qualified)
+
+to a ``dune`` file, Dune will consider all the modules in the directory and its
+subdirectories, adding a prefix to the module name in the usual Coq style for
+subdirectories. For example, file ``A/b/C.v`` will be module ``A.b.C``.
+
+Limitations
+~~~~~~~~~~~
+
+- ``.v`` files always depend on the native version of Coq / plugins,
```

Comment:
```suggestion
- ``.v`` files always depend on the native version of Coq / plugins,
```
@ejgallego Would you mind explaining this?

Reply:
I mean that we always use `ltac_plugin.cmxs` and `coqc` instead of `ltac_plugin.cmo` and `coqc.byte`.

We could add a switch for this, not worth it IMO.

Reply:
OK I'll make that comment a bit clearer, but it is mostly an implementation detail of the plugin on the Coq side rather than something the dune users have to be aware of right.

Reply:
Yes, tho some users may care that dune calls `coqc.byte` ? I don't think so, so indeed we can update the comment to say that "Dune works exclusively with the native-version of Coq, except when not available"

Tho that's a bit confusing... most people won't know what native version of Coq is, and moreover we got a problem with the confusion w.r.t. native compiler in Coq (which targets the OCaml native compiler)

Reply:
I will omit it for now.